src/pip/_internal/req/req_install.py

```diff
@@ -443,7 +445,9 @@ def pyproject_toml(self):
 
         # Python2 __file__ should not be unicode
         if six.PY2 and isinstance(pp_toml, six.text_type):
-            pp_toml = pp_toml.encode(sys.getfilesystemencoding())
+            fs_enc = sys.getfilesystemencoding()
+            if fs_enc is not None:
+                pp_toml = pp_toml.encode(fs_enc)
 
         return pp_toml
 
```

Comment:
Same here - `pp_toml` needs to have a consistent type (from the comment, bytes).